packages/settingeditor-extension/src/settingeditor.ts

```diff
@@ -791,10 +892,47 @@ class PluginFieldset extends Widget {
 
     // Populate if possible.
     if (settings) {
-      Private.populateFieldset(this.node, settings.plugin, settings.schema);
+      Private.populateFieldset(this.node, settings);
+    }
+  }
+
+  /**
+   * Handle the `'click'` event for the plugin fieldset.
+   *
+   * @param event - The DOM event sent to the widget
+   */
+  private _evtClick(event: MouseEvent): void {
+    const attribute = 'data-property';
+    const root = this.node;
+    let target = event.target as HTMLElement;
+
+    while (target && target.parentElement !== root) {
+      if (target.hasAttribute(attribute)) {
+        event.preventDefault();
+        this._onPropertyAdded(target.getAttribute(attribute));
+        return;
+      }
+      target = target.parentElement;
     }
   }
 
+  /**
+   * Handle a property addition.
+   */
+  private _onPropertyAdded(property: string): void {
+    const settings = this._settings;
+
+    settings.save({ ...settings.user, [property]: settings.default(property) })
```

Comment:
I love the expressiveness of the spread operator and the ability to dynamically index into objects; it's nicer than plain old JS.